pkg/workload/admissionchecks.go

```diff
@@ -99,6 +99,18 @@ func ResetChecksOnEviction(w *kueue.Workload) {
 	}
 }
 
+func ResetChecksOnDeactivation(w *kueue.Workload) {
+	checks := w.Status.AdmissionChecks
+	for i := range checks {
+		checks[i] = kueue.AdmissionCheckState{
+			Name:               checks[i].Name,
+			State:              kueue.CheckStatePending,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+			Message:            "Reset to Pending after deactivation. Previously: " + string(checks[i].State),
```

Comment:
This will unnecessarily change the message for a Pending admission check. It looks harmless but unnecessary.
I would suggest to skip updating Pending checks, and return `updated` from this function if any AC state was updated.